### ISSUE TITLE
KeyError in get_budgets with budget in top-level category

### ISSUE THREAD

**Issue by LubergAlexander** [mrooney/mintapi#36]
```python
mint.get_budgets()

/Library/Python/2.7/site-packages/mintapi/api.pyc in get_budgets(self)
    254         for direction in budgets.keys():
    255             for budget in budgets[direction]:
--> 256                 budget['cat'] = categories[budget['cat']]
    257
    258         return budgets

KeyError: 13
```

**Comment by mark-adams** [comment]
I'm not able to duplicate this issue. `mint.get_budgets()` works for me.

**Comment by LubergAlexander** [comment]
ok! I will try to debug it today, and probably will send a pull request

**Comment by LubergAlexander** [comment]
The root cause is that the lookup is done only on the first level. Categories can be nested. Since you are using the https://wwws.mint.com/bundledServiceController.xevent, the tree is reversed(from child to parents).

There are a few options: 1) to implement tree traversal, and lookup ids in parent fields(which can be a few levels deep)
2) to switch to another endpoint: https://wwws.mint.com/app/getJsonData.xevent?task=categories

This endpoint returns parent nodes with children, which is a list, thus the traversal is quite easy & quick.

You can take a look here: https://github.com/KartikTalwar/mint/blob/28802d5d416bafeabbaa51d3f5da4e5dabff424d/mint/__init__.py

Here is the implementation of the lookup including children
```python

  def get_category_from_id(self, cid):
    if cid == 0:
      return 'Uncategorized'

    for i in self.get_categories():
      if i['id'] == cid:
        return i['value']
      if 'children' in i:
        for j in i['children']:
          if j['id'] == cid:
            return j['value']

    return 'Unknown'
```

**Comment by bbusenius** [comment]
I can confirm this error. I'm getting the same thing.

**Comment by mcronce** [comment]
Finally looked into this today (my fiancee's account was getting the error and polluting my mail on the box running the script).  It's caused by having a budget against a top-level category; if you only have budgets against the child categories, you won't experience it.

With that cause in mind, I'll get a fix in and submit a pull request...sometime when I've got the time available.

**Comment by mrooney** [comment]
Thanks for debugging, I updated the title! I'll happily review and merge the PR :)

**Comment by graham1776** [comment]
Is there any update on this? Would love to see categories fixed

**Comment by mrooney** [comment]
@mcronce if you (or anyone else) happened to end up fixing this locally, just send along a PR :)

**Comment by tylerneu** [comment]
[PR submitted.](https://github.com/mrooney/mintapi/pull/46)

**Comment by graham1776** [comment]
@mrooney Mind merging?

**Comment by mcronce** [comment]
I have had exactly zero time to do any coding :( that said, I glanced over the changed in the PR, and it certainly looks good to me.

**Comment by graham1776** [comment]
You guys are awesome. Used the new master and the categories worked perfectly. Thanks again!

**Comment by mrooney** [comment]
Thanks all, released to pypi as 1.11!
